src/ray/object_manager/plasma/connection.cc

```diff
@@ -98,11 +101,16 @@ std::shared_ptr<Client> Client::Create(PlasmaStoreMessageHandler message_handler
         }
       };
 
-  auto self =
-      std::make_shared<Client>(PrivateTag{}, ray_message_handler, std::move(socket));
-  // Let our manager process our new connection.
-  self->ProcessMessages();
-  return self;
+  ray::ConnectionErrorHandler ray_connection_error_handler =
+      [connection_error_handler](std::shared_ptr<ray::ClientConnection> client,
+                                 const boost::system::error_code &error) {
+        connection_error_handler(
+            std::static_pointer_cast<Client>(client->shared_ClientConnection_from_this()),
```

Comment:
not sure why do you need to get shared pointer from this; 
isn't (1) client; and (2) client->shared_from_this both of type shared_ptr<ClientConnection>?

Reply:
was existing code, no idea why it was written that way, removed the `shared_from_this`